### PR TITLE
Added local field and number of excitons for S1 time generation

### PR DESCRIPTION
**Pull request by terliuk**
**What is the problem / what does the code in this PR do**
This PR adds a possibility to store local field and number of excitons to instructions. This information is needed for usage of S1 pulse shape coming from NEST and Geant4 timing splines.

### CODE REVIEW

File: epix/io.py [hunk 1]
```diff
@@ -227,15 +227,18 @@ def awkward_to_wfsim_row_style(interactions):
         res['g4id'][i::2] = awkward_to_flat_numpy(interactions['evtid'])
         res['vol_id'][i::2] = awkward_to_flat_numpy(interactions['vol_id'])
         res['e_dep'][i::2] = awkward_to_flat_numpy(interactions['ed'])
-        
+        if 'local_field' in res.dtype.names:
```
Comment (HenningSE) [inline epix/io.py:230]: Why do we need the if case here? `res.dtype` is taken from WFsim defined [here](https://github.com/XENONnT/WFSim/blob/master/wfsim/strax_interface.py#L31-L32) I think so the condition is always true?

File: epix/io.py [hunk 2]
```diff
@@ -227,15 +227,18 @@ def awkward_to_wfsim_row_style(interactions):
...
+
         recoil = awkward_to_flat_numpy(interactions['nestid'])
         res['recoil'][i::2] = np.where(np.isin(recoil, [0,6,7,8,11]), recoil, 8)
 
         if i:
             res['amp'][i::2] = awkward_to_flat_numpy(interactions['electrons'])
         else:
             res['amp'][i::2] = awkward_to_flat_numpy(interactions['photons'])
-
+            if 'n_excitons' in res.dtype.names:
```
Comment (HenningSE) [inline epix/io.py:240]: Same question as for `local_field`

**Review by HenningSE — COMMENTED** [review]
Hi Andrii, nice PR. Just one quick question and then I think it is ready.

**Comment by ramirezdiego** [comment]
I agree with @HenningSE that this is the only open point. If the reason to keep these is backwards compatibility with older WFSim versions, I suggest we just remove them. It's good to always aim at backwards compatibility, but so far there aren't blessed epix+WFSim runs that we may want to reproduce in the future (please correct me if I am wrong).

File: epix/io.py (hunk 1 above)
Comment (terliuk) [reply epix/io.py:230]: It is backwards compatibility, when epix is used with older version of WFsim. Probably not really necessary, but I prefer to keep compatibility for +- few versions of the code.

**Comment by terliuk** [comment]
oops :) did not notice reply from Diego

**Review by HenningSE — APPROVED** [review]
Ok, I have no strong opinion on the backward compatibility. We can merge I think (or remove the if case and then merge however you prefer)

**Review by ramirezdiego — APPROVED** [review]
Ok, for me too. @terliuk prefers to keep the if statements, so let's reward him!

**Pull request merged**